src/GojsDiagram.test.tsx

```diff
@@ -0,0 +1,150 @@
+                    go.TextBlock,
+                    { margin: 8 },
+                    new go.Binding('text', 'key'))
+            );
+
+        return myDiagram;
+    };
+
+    const myDiagramId = 'myDiagramId';
+
+    let diagram: Diagram;
+    let wrapper;
+
+    beforeEach(() => {
+        const dom = document.body;
+        wrapper = mount(
+            (
+                <GojsDiagram
+                    diagramId={myDiagramId}
+                    model={model}
+                    createDiagram={(id) => {
+                        diagram = createDiagram(myDiagramId);
+                        return diagram;
+                    }
+                    }
+                    className="fakecss"
+                />
+            ),
+            { attachTo: dom });
+    });
+
+    it('renders the model provided as parameter', () => {
+        checkRenderedModel(model);
+        wrapper.unmount();
+    });
+
+    const checkRenderedModel = (currentModel) => {
+        expect(diagram.nodes.count).toBe(currentModel.nodeDataArray.length);
+        currentModel.nodeDataArray.forEach(node => {
+            expect(diagram.nodes.any(e => e.key === node.key && e.data.color === node.color)).toBeTruthy();
+        });
+        currentModel.linkDataArray.forEach(link => {
+            expect(diagram.links.any(e => e.fromNode.key === link.from && e.toNode.key === link.to)).toBeTruthy();
+        });
+    };
+
+    const testInputs = [
+        {
+            updatedModel: {
+                ...model,
+                nodeDataArray: [
+                    ...model.nodeDataArray,
+                    { key: 'New', color: 'blue' }
+                ]
+            }
+        },
+        {
+            updatedModel: {
+                nodeDataArray: [
+                    { key: 'Alpha', color: 'lightblue' },
+                    { key: 'Beta', color: 'orange' },
+                    { key: 'Gamma', color: 'lightgreen' },
+                ],
+                linkDataArray:
+                    [
+                        { from: 'Alpha', to: 'Beta' },
+                        { from: 'Alpha', to: 'Gamma' }
+                    ]
+            }
+        },
+        {
+            updatedModel: {
+                ...model,
+                linkDataArray:
+                    [
+                        ...model.linkDataArray,
+                        { from: 'Alpha', to: 'Omega' }
+                    ]
+            }
+        },
+        {
+            updatedModel: {
+                ...model,
+                nodeDataArray: [
+                    { key: 'Alpha', color: 'lightblue' },
+                    { key: 'Beta', color: 'blue' },
+                    { key: 'Gamma', color: 'black' },
+                    { key: 'Delta', color: 'pink' },
+                    { key: 'Omega', color: 'red' }
+                ]
+            }
+        }
+    ];
+
+    testInputs.forEach(test => {
+        it('updates the render given a new model', () => {
```

Comment:
is it possible to unroll this loop and name each test case accordingly? 
this coding style is very hard to read.... only the person that wrote the test may know what each test case is actually looking for...

if you really want to test it this way... I would recomment:

1. adding a text field to each update to describe what you are looking for on each input
2. rename  `testInputs`  to `testCases`
3. Use the text field to give a different parameter for `it(...)` on each iteration

Reply:
I've just added a description property in the test case. This property is used in the definition of the it method. So when you run the test, you have a clear description of each test.